lib/core-integration/src/Test/Integration/Scenario/CLI/Shelley/HWWallets.hs

```diff
@@ -303,46 +305,39 @@ spec = describe "SHELLEY_CLI_HW_WALLETS" $ do
             code `shouldBe` ExitSuccess
 
     describe "HW_WALLETS_05 - Wallet from pubKey is available" $ do
-        it "The same account and mnemonic wallet can live side-by-side" $ \ctx -> do
+        it "The same account and mnemonic wallet can live side-by-side" $ \ctx -> runResourceT $ do
             Stdout m <- generateMnemonicsViaCLI @t []
             let pubKeyWalName = "pub key wallet"
             let mnemonicWalName = "mnemonic wallet"
             -- create wallet from mnemonics
             (c1, o1, e1) <- createWalletViaCLI @t ctx [mnemonicWalName] m "\n" "secure-passphrase"
             c1 `shouldBe` ExitSuccess
             T.unpack e1 `shouldContain` cmdOk
-            _ <- expectValidJSON (Proxy @ApiWallet) o1
+            mnemonicWal <- expectValidJSON (Proxy @ApiWallet) o1
 
             -- create wallet from pub key
             let accXPub = pubKeyFromMnemonics' (words m)
             (Exit c2, Stdout o2, Stderr e2) <-
                 createWalletFromPublicKeyViaCLI @t ctx [pubKeyWalName, accXPub]
             c2 `shouldBe` ExitSuccess
             e2 `shouldContain` cmdOk
-            _ <- expectValidJSON (Proxy @ApiWallet) o2
+            pubKeyWal <- expectValidJSON (Proxy @ApiWallet) o2
 
             (Exit c, Stdout out, Stderr err) <- listWalletsViaCLI @t ctx
+            wids <- map (view walletId) <$> expectValidJSON (Proxy @[ApiWallet]) out
             c `shouldBe` ExitSuccess
             err `shouldBe` cmdOk
-            rl <- expectValidJSON (Proxy @[ApiWallet]) out
-            length rl `shouldBe` 2
-            verify rl
-                [ expectCliListField 0
-                    (#name . #getApiT . #getWalletName)
-                    (`shouldBe` T.pack mnemonicWalName)
-                , expectCliListField 1
-                    (#name . #getApiT . #getWalletName)
-                    (`shouldBe` T.pack pubKeyWalName)
-                ]
+            wids `shouldContain` [mnemonicWal ^. walletId]
```

Comment:
Now checking that the wallet ids of the listed wallets _contain_ these two, instead of asserting that the first and second wallet in the list have the right names.